libs/common/src/helpers/schema/z-boolean-strict-coerce.ts

```diff
@@ -0,0 +1,24 @@
+import z from 'zod';
+
+const booleanLike = z.union([
```

Comment:
Do we even need `zod` package, can we not do this natively?

Reply:
I'd say zod offers a lot more than us hand rolling validation ourselves (our existing validation has a few blind spots -- which zod covers nicely).

It also provides good error msg reporting, and is easy to use and read.

It also allows for type inference from schemas, meaning once we add validation to something, we also can generate a typescript type from it (one source of truth):
```
const settingsSchema = z.object({
  darkMode: z.boolean().default(false),
  notifications: z.boolean().default(true),
});

type Settings = z.infer<typeof settingsSchema>;
```

tldr - I think it offers a massive timesave into us rolling our own validation, with some great benefits like type inference.